tests/phpunit/CRM/Case/BAO/QueryTest.php

```diff
@@ -116,4 +116,62 @@ public function testFindCasesQuery() {
     );
   }
 
+  /**
+   * Tests the advanced search query by searching on related contacts and case type at the same time.
+   *
+   * Preparation:
+   *   Create a contact Contact A
+   *   Create another contact Contact B
+   *   Create a third contact Contact C
+   *   Create a case of type Housing Support for Contact A
+   *   On the case assign the role Benefit specialist is to Contact B
+   *   Create a second case of type Adult day care referral
+   *   On the case assign the role Benefit specialist is to Contact C
+   *
+   * Searching:
+   *   Go to advanced search
+   *   Click on View contact as related contact
+   *   Select Benefit Specialist as relationship type
+   *   Go to tab cases and select Housing Support as case type
+   *
+   * Expected results
+   *   We expect to find contact B and not C.
+   *
+   * @throws \Exception
+   */
+  public function testAdvancedSearchWithDisplayRelationshipsAndCaseType() {
+    // Preperation
+    $benefitRelationshipTypeId = civicrm_api3('RelationshipType', 'getvalue', ['return' => 'id', 'name_a_b' => 'Benefits Specialist is']);
+    $clientContactID = $this->individualCreate(['first_name' => 'John', 'last_name' => 'Smith']);
+    $benefitSpecialist1 = $this->individualCreate(['Individual', 'first_name' => 'Alexa', 'last_name' => 'Clarke']);
+    $benefitSpecialist2 = $this->individualCreate(['Individual', 'first_name' => 'Sandra', 'last_name' => 'Johnson']);
+    $housingSupportCase = $this->createCase($clientContactID, NULL, ['case_type_id' => 'housing_support', 'case_type_id' => 1]);
```

Comment:
I think this is a typo: case_type_id used twice.

Reply:
Yes indeed is a typo